pkg/ibmvpc/parser.go

```diff
@@ -150,13 +150,132 @@ func VPCConfigsFromResources(rc *datamodel.ResourcesContainerModel, vpcID, resou
 		return nil, err
 	}
 
+	err = getRoutingTables(rc, res)
+	if err != nil {
+		return nil, err
+	}
+
 	if debug {
 		printVPCConfigs(res)
 	}
 
 	return res, nil
 }
 
+// getRoutingTables parses routing tables from rc and adds their generated objects to
+// the relevant vpc configs within res
+func getRoutingTables(rc *datamodel.ResourcesContainerModel,
+	res *vpcmodel.MultipleVPCConfigs) error {
+	for _, rt := range rc.RoutingTableList {
+		routes, err := getRoutes(rt)
+		if err != nil {
+			return err
+		}
+		if rt.VPC == nil || rt.VPC.CRN == nil {
+			logging.Debugf("unknown vpc for rt %s, skipping...", *rt.Name)
+			continue
+		}
+		vpcUID := *rt.VPC.CRN
+		vpcConfig := res.Config(vpcUID)
+		if vpcConfig == nil {
+			logging.Debugf("skipping rt %s, could not find vpc with uid %s", *rt.Name, vpcUID)
+			fmt.Printf("skipping rt %s, could not find vpc with uid %s\n", *rt.Name, vpcUID)
+			continue
+		}
+		var rtObj vpcmodel.VPCResourceIntf
+		if *rt.RouteDirectLinkIngress || *rt.RouteInternetIngress || *rt.RouteTransitGatewayIngress || *rt.RouteVPCZoneIngress {
+			rtObj, err = getIngressRoutingTable(rt, routes, vpcConfig)
+		} else {
+			rtObj, err = getEgressRoutingTable(rt, routes, vpcConfig)
+		}
+		if err != nil {
+			return err
+		}
+		logging.Debugf("add rt %s for vpc %s\n", rtObj.Name(), vpcUID)
+
+		vpcConfig.AddRoutingTable(rtObj)
+		res.SetConfig(vpcUID, vpcConfig)
+	}
+	return nil
+}
+
+func getIngressRoutingTable(rt *datamodel.RoutingTable,
+	routes []*route,
+	vpcConfig *vpcmodel.VPCConfig) (vpcmodel.VPCResourceIntf, error) {
+	if !*rt.RouteTransitGatewayIngress {
+		// todo: skip instead?
+		return nil, fmt.Errorf("currently not supporting ingress routing table with source other than tgw ")
+	}
+	res := newIngressRoutingTableFromRoutes(routes, vpcConfig)
+	res.ResourceName = *rt.Name
+	res.ResourceUID = *rt.ID
+	res.ResourceType = ResourceTypeRoutingTable
+	res.VPCRef = vpcConfig.VPC
+	return res, nil
+}
+
+func getEgressRoutingTable(rt *datamodel.RoutingTable,
+	routes []*route,
+	vpcConfig *vpcmodel.VPCConfig) (vpcmodel.VPCResourceIntf, error) {
+	subnets := []*Subnet{}
+	for _, s := range rt.Subnets {
+		if sObj, ok := vpcConfig.UIDToResource[*s.CRN]; ok {
+			if subnet, ok := sObj.(*Subnet); ok {
+				subnets = append(subnets, subnet)
+			}
+		} else {
+			return nil, fmt.Errorf("could not find subnet %s associated with routing table %s", *s.Name, *rt.Name)
+		}
+	}
+
+	res := newEgressRoutingTableFromRoutes(routes, subnets, vpcConfig)
+	res.ResourceName = *rt.Name
+	res.ResourceUID = *rt.ID
+	res.ResourceType = ResourceTypeRoutingTable
+	res.VPCRef = vpcConfig.VPC
```

Comment:
consider setting some of these fields in `newEgressRoutingTableFromRoutes()`

Reply:
done